rspace/src/main/scala/coop/rchain/rspace/history/package.scala

```diff
@@ -324,28 +320,24 @@ package object history {
           // If there are is only one child, then we know that it is the thing we are trying to
           // propagate upwards, and we can go ahead and do that.
           // post skip node: this case got swallowed in deleteLeaf
-          case Vector(_) =>
-            propagateTrieUpward(store, txn, ptr, ByteVector(byte), tail)
+          case Vector(_) => propagateTrieUpward(ptr, ByteVector(byte), tail)
```

Comment:
Makes me wonder why this case isn't covered

Reply:
It is in `deleteLeaf`. As in `propagateTrieUpward` will never be called by `deleteLeaf` with a singular child because it would be an invalid Trie that has Skips.
IOW the structure of the Trie with Skips does not allow 1 element PointerBlocks.
The case is valid in abstract so I didn't want to remove it. Do you think it should go the way of the dodo?

Reply:
Codecov says it isn't covered. Do we have a test for that?

Reply:
I agree with @lukasz-golebiewski - we need to get to the bottom of this.  It isn't these changes that introduced this situation though, so I gather we missed it before.  I'm okay with investigating that, making changes, and/or adding tests in a separate PR, but let's be sure to make that work a priority.